crates/ruff_linter/src/message/text.rs

```diff
@@ -116,14 +116,16 @@ impl Emitter for TextEmitter {
             )?;
 
             if self.flags.intersects(EmitterFlags::SHOW_SOURCE) {
-                writeln!(
-                    writer,
-                    "{}",
-                    MessageCodeFrame {
-                        message,
-                        notebook_index
-                    }
-                )?;
+                if message.range().len() > TextSize::default() {
```

Comment:
I think we should probably restrict it to `0..0` as this will also catch any `n..n` where `n` > 0.

```suggestion
				// The `0..0` range is used to highlight file-level diagnostics.
                if message.range() != TextRange::default() {
```